File: xamarin/Samples/Showcase-iOS/ViewController/PdfDisplayViewController.cs
using Foundation;
using LiferayScreens;
using System;
using UIKit;

namespace ShowcaseiOS.ViewController
{
    public partial class PdfDisplayViewController : UIViewController, IFileDisplayScreenletDelegate
    {
        public PdfDisplayViewController(IntPtr handle) : base(handle) { }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            this.pdfDisplayScreenlet.ClassPK = LiferayServerContext.LongPropertyForKey("pdfClassPK");
            this.pdfDisplayScreenlet.ClassName = LiferayServerContext.StringPropertyForKey("fileClassName");

            this.pdfDisplayScreenlet.Delegate = this;
        }

        /* IFileDisplayScreenletDelegate */

        [Export("screenlet:onFileAssetError:")]
        public virtual void OnFileAssetError(FileDisplayScreenlet screenlet, NSError error)
        {
            Console.WriteLine($"Pdf display failed: {error.DebugDescription}");
        }

        [Export("screenlet:onFileAssetResponse:")]
        public virtual void OnFileAssetResponse(FileDisplayScreenlet screenlet, NSUrl url)
        {
            Console.WriteLine($"Pdf display success: {url.DebugDescription}");
        }
    }
}

File: ios/Samples/Showcase-ObjectiveC/Showcase-ObjectiveC/ViewControllers/Files/PdfDisplayViewController.m
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */


@import LiferayScreens;
#import "LiferayLogger.h"
#import "PdfDisplayViewController.h"

@interface PdfDisplayViewController () <FileDisplayScreenletDelegate>

@property (weak, nonatomic) IBOutlet PdfDisplayScreenlet *screenlet;

@end

@implementation PdfDisplayViewController

- (void)viewDidLoad {
	[super viewDidLoad];

	self.screenlet.delegate = self;
	self.screenlet.presentingViewController = self;
	self.screenlet.classPK = [LiferayServerContext longPropertyForKey:@"pdfDisplayClassPK"];
}

#pragma mark FileDisplayScreenletDelegate

- (void)screenlet:(FileDisplayScreenlet *)screenlet onFileAssetError:(NSError *)error {
	LiferayLog(error.debugDescription);
}

- (void)screenlet:(FileDisplayScreenlet *)screenlet onFileAssetResponse:(NSURL *)url {
	LiferayLog(url);
}

@end
